homeassistant/components/hassio/websocket_api.py

```diff
@@ -36,6 +41,33 @@ def async_load_websocket_api(hass: HomeAssistant):
     """Set up the websocket API."""
     websocket_api.async_register_command(hass, websocket_supervisor_event)
     websocket_api.async_register_command(hass, websocket_supervisor_api)
+    websocket_api.async_register_command(hass, websocket_subscribe)
+
+
+@websocket_api.require_admin
+@websocket_api.async_response
+@websocket_api.websocket_command({vol.Required(WS_TYPE): WS_TYPE_SUBSCRIBE})
+async def websocket_subscribe(
+    hass: HomeAssistant, connection: ActiveConnection, msg: dict
+):
+    """Subscribe to supervisor events."""
+
+    async def forward_messages(data):
+        """Forward events to websocket."""
+        connection.send_message(websocket_api.event_message(msg[WS_ID], data))
+
+    subscriptions = {
+        EVENT_SUPERVISOR_EVENT: async_dispatcher_connect(
+            hass, EVENT_SUPERVISOR_EVENT, forward_messages
+        ),
+    }
+
+    @callback
+    def unsubscribe():
+        subscriptions[EVENT_SUPERVISOR_EVENT]()
+
+    connection.subscriptions[msg[WS_ID]] = unsubscribe
```

Comment:
This works too :)

```suggestion
    connection.subscriptions[msg[WS_ID]] = async_dispatcher_connect(
        hass, EVENT_SUPERVISOR_EVENT, forward_messages
    )
```

Reply:
It does? Tests locked up when I tried it